pkg/hubble/metrics/metrics_test.go

```diff
@@ -46,3 +54,66 @@ func TestInitializedMetrics(t *testing.T) {
 	})
 
 }
+
+func ConfigureAndFetchMetrics(t *testing.T, testName string, metricCfg []string, exportedMetrics map[string][]string) {
+	t.Run(testName, func(t *testing.T) {
+		log := logrus.New()
+		grpcMetrics := grpc_prometheus.NewServerMetrics()
+		EnableMetrics(
+			prometheus.NewPedanticRegistry(),
+			log,
+			":6767",
+			nil,
+			metricCfg,
+			grpcMetrics,
+			false)
+
+		time.Sleep(200 * time.Millisecond)
```

Comment:
Not ideal, but I think we can always do a follow up later to make it possible to pass our own `http.Server` so we can use something like `httptest.Server` instead. As long as this doesn't flake much I think this is fine.

Reply:
Sounds good!

Reply:
Sure, I could use a pointer: I tried using a channel to set a 'ready' flag from within initMetricsServer, but that's not early enough - seems it would have to happen within the l.Accept() call in Serve() in server.go, but it can't be modified being part of Go standard library itself?

Let me look into setting up the server in the test and passing it into EnableMetrics in the meantime

Reply:
And if it's a mock, doesn't that defeat the purpose of this being more of an E2E test in preparation for the changes to metrics cardinality work, since there already are pure unit tests for metrics?

Reply:
I would refactor all of the functions related to metrics servers, `EnableMetrics`, `initMetrics`, `initMetricsServer`, etc to take an `http.Server` and we can test with an `httptest.Server` instead. In `launchHubble` we can construct the `http.Server` instead, along with the TLS config, etc, and we'll call `EnableMetrics` in a Go routine, making the caller responsible for running it in a Go routine instead.

Reply:
I ended up splitting server creation out altogether. Since EnableMetrics was going to call the blocking ListenAndServe both for product and test code, unless I passed in an ugly flag branching the test scenario within the initMetricsServer.